### PR TITLE
Simplify `make_evaluation_predictions`

### PR DESCRIPTION
**Pull request by lostella**
*Description of changes:* I think this brings `make_evaluation_predictions` closer to deprecation, since its behavior is very simple in the end.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/evaluation/backtest.py [hunk 1]
```diff
@@ -30,7 +32,20 @@
...
 from gluonts.model.predictor import Predictor
 from gluonts.itertools import maybe_len
-from gluonts.transform import AdhocTransform
+
+
+def _to_dataframe(input_label: Tuple[DataEntry, DataEntry]) -> pd.DataFrame:
+    """
+    Turn a pair of consecutive (in time) data entries into a dataframe.
+    """
+    targets = tuple(entry[FieldName.TARGET] for entry in input_label)
```
Comment (jaheba) [inline src/gluonts/evaluation/backtest.py]: Is there a difference between `tuple` and `list here`?

File: src/gluonts/evaluation/backtest.py [hunk 2]
```diff
@@ -30,7 +32,20 @@
...
+    target = np.concatenate(targets, axis=-1)
+    index = pd.period_range(
```
Comment (jaheba) [inline src/gluonts/evaluation/backtest.py]: We really should have a helper for this.

File: src/gluonts/evaluation/backtest.py (hunk 1 above)
Comment (lostella) [reply src/gluonts/evaluation/backtest.py]: Probably not

**Comment by lostella** [comment]
Turns out that the multivariate case needed some attention, so I added some fixes here, which I can move to a separate PR if necessary.

Still to do: fix a test with LSTNet, where `lead_time > 0` is creating issues

File: src/gluonts/evaluation/backtest.py (hunk 2 above)
Comment (lostella) [reply src/gluonts/evaluation/backtest.py]: Agreed

File: src/gluonts/dataset/rolling_dataset.py [hunk 3]
```diff
@@ -111,7 +111,7 @@ def truncate_features(timeseries: dict, max_len: int) -> dict:
     ):
         if key not in timeseries:
             continue
-        timeseries[key] = [feature[:max_len] for feature in timeseries[key]]
+        timeseries[key] = timeseries[key][:, :max_len]
```
Comment (jaheba) [inline src/gluonts/dataset/rolling_dataset.py]: Unrelated change?

File: src/gluonts/dataset/util.py [hunk 4]
```diff
@@ -14,21 +14,32 @@
...
 
 def forecast_start(entry, time_axis: int = -1):
     return entry[FieldName.START] + entry[FieldName.TARGET].shape[time_axis]
 
 
-def to_pandas(instance: dict, freq: Optional[str] = None) -> pd.Series:
+def period_index(entry: DataEntry, freq=None) -> pd.PeriodIndex:
+    if freq is None:
+        freq = entry[FieldName.START].freq
+    return pd.period_range(
```
Comment (jaheba) [inline src/gluonts/dataset/util.py:29]:
```suggestion

    return pd.period_range(
```

**Review by jaheba — DISMISSED** [review]

**Review by jaheba — APPROVED** [review]
